apps/src/p5lab/spritelab/commands/validationCommands.js

```diff
@@ -57,10 +64,583 @@ export const commands = {
+      ) {
+        foundEventSpriteChange = true;
+      } else if (!utils.arrayEquals(currentBehaviors, previousBehaviors)) {
+        foundNoneventSpriteChange = true;
+      }
+    }
+    result = foundEventSpriteChange && !foundNoneventSpriteChange;
+    return result;
+  },
+
+  // Returns true if all sprites have the default size (100 for students)
+  spritesDefaultSize() {
+    const spriteIds = this.getSpriteIdsInUse();
+    let result = true;
+    for (let i = 0; i < spriteIds.length; i++) {
+      if (this.nativeSpriteMap[spriteIds[i]].getScale() !== 1) {
+        result = false;
+      }
+    }
+    return result;
+  },
+
+  // Returns true if any sprite was clicked, regardless of the eventLog.
+  anySpriteClicked() {
+    const spriteIds = this.getSpriteIdsInUse();
+    let result = false;
+    if (this.p5.mouseWentDown('left')) {
+      for (let i = 0; i < spriteIds.length; i++) {
+        if (this.p5.mouseIsOver(this.nativeSpriteMap[spriteIds[i]])) {
+          result = true;
+        }
+      }
+    }
+    return result;
+  },
+
+  // Returns true if any two sprites touched, regardless of the eventLog.
+  anySpritesTouched() {
+    let result = false;
+    const allSprites = this.p5.World.allSprites;
+    result = allSprites.isTouching(allSprites);
+    return result;
+  },
+
+  // Returns true if a click event was logged this frame.
+  clickEventFound() {
+    let result = false;
+
+    //Only check for values that are new this frame
+    for (let i = this.previous.eventLogLength; i < this.eventLog.length; i++) {
+      if (
+        this.eventLog[i].includes('whenClick: ') ||
+        this.eventLog[i].includes('whileClick: ')
+      ) {
+        result = true;
+      }
+    }
+
+    return result;
+  },
+
+  // Returns true if a touch event was logged this frame.
+  touchEventFound() {
+    let result = false;
+
+    //Only check for values that are new this frame
+    for (let i = this.previous.eventLogLength; i < this.eventLog.length; i++) {
+      if (
+        this.eventLog[i].includes('whenTouch: ') ||
+        this.eventLog[i].includes('whileTouch: ')
+      ) {
+        result = true;
+      }
+    }
+
+    return result;
+  },
+
+  // Returns true if a time event was logged this frame.
+  atTimeEventFound() {
+    let result = false;
+
+    //Only check for values that are new this frame
+
+    for (let i = this.previous.eventLogLength; i < this.eventLog.length; i++) {
+      if (this.eventLog[i].includes('atTime: ')) {
+        result = true;
+      }
+    }
+    return result;
+  },
+
+  // Returns true text was printed this frame.
+  printedText() {
+    const previousPrintLogLength = this.previous.printLogLength || 0;
+    let result = previousPrintLogLength < this.printLog.length;
+    return result;
+  },
+
+  // Returns true if the student set a variable to any number, string, or boolean.
+  variableCreated() {
+    let result = this.studentVars.length >= 1;
+    return result;
   }
 };
+class criteria {
```

Comment:
This might be a good place to document the fields and even give some examples of each.